src/cmd/exec.rs

```diff
@@ -14,14 +16,35 @@ fn run_in_context(kubeconfig: &KubeConfig, args: &[String]) -> anyhow::Result<i3
     let depth = vars::get_depth();
     let next_depth = depth + 1;
 
-    let mut proc = Command::new(&args[0])
+    let mut child = Command::new(&args[0])
         .args(&args[1..])
         .env("KUBECONFIG", temp_config_file.path())
         .env("KUBIE_KUBECONFIG", temp_config_file.path())
         .env("KUBIE_ACTIVE", "1")
         .env("KUBIE_DEPTH", next_depth.to_string())
         .spawn()?;
-    let status = proc.wait()?;
+
+    let child_pid = child.id();
+
+    thread::spawn(move || {
+        let signals = Signals::new(&[
```

Comment:
Signal handlers should be installed before the process is spawned to avoid race conditions.